vscode-extension/src/aiConfigEditor.ts

```diff
@@ -199,17 +234,17 @@ export class AIConfigEditorProvider implements vscode.CustomTextEditorProvider {
 
     // Make sure we get rid of the listener when our editor is closed.
     webviewPanel.onDidDispose(() => {
+      isWebviewDisposed = true;
       console.log(`${document.fileName}: Webview disposed`);
+
       changeDocumentSubscription.dispose();
       willSaveDocumentSubscription.dispose();
 
-      // TODO: saqadri -- terminate the editor server process.
       if (editorServer) {
         console.log("Killing editor server process");
         editorServer.proc.kill();
+        editorServer = null;

```

Comment:
Why do we need to explicitly set this to null? Is this related to the logic in the rest of this PR or just finishing the TODO from Sarmad on L206?

Reply:
Honestly, I'm not sure if it needs to be (seems same behaviour with or without) but the fact that the existing code has webViewDisposed check and editorServer check suggests we *should* be setting it to null when its process is killed. More for completeness than anything